### PR TITLE
Porting C++ Test Agent to 1.0.1-RC1

### LINKED ISSUE #59: Update up-tck C++ test agent for 1.5.8 API

### PR DESCRIPTION
**Pull request by agosh01**
This PR update c++ test agent to work with 1.6.0 core api with addition on L2 apis (closes #59 )

1. Updating to 1.6.0
2. Refactoring 
3. Feature file updates

### CODE REVIEW

File: test_agent/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -12,13 +12,18 @@
...
 project(test_agent_cpp VERSION 0.1  LANGUAGES CXX DESCRIPTION "C++ Test Agent")
 
+find_package(protobuf REQUIRED)
+find_package(up-core-api REQUIRED)
 find_package(up-cpp REQUIRED)
 find_package(spdlog REQUIRED)
 find_package(fmt REQUIRED CONFIG)
 find_package(RapidJSON CONFIG REQUIRED)
-find_package(up-client-zenoh-cpp REQUIRED)
+# find_package(up-client-zenoh-cpp REQUIRED)
```
Comment (agosh01) [inline test_agent/cpp/CMakeLists.txt:21]: Add TODO for zenoh

File: test_agent/cpp/include/utils.h [hunk 2]
```diff
@@ -0,0 +1,81 @@
+#ifndef UTILS_H
```
Comment (matthewd0123) [inline test_agent/cpp/include/utils.h:1]: Add license to this file?

**Review by matthewd0123 — COMMENTED** [review]
Honestly only one thing I found... not the best guy for reviewing C++ but it all looks good from me. Feature files are looking great

**Review by matthewd0123 — APPROVED** [review]
LGTM

**Review by billpittman — CHANGES_REQUESTED** [review]
Only a few comments/questions.

**Review by gregmedd — APPROVED** [review]

File: test_manager/reports/summary/summary.json
[the remark this answers is not in the document]
Comment (agosh01) [reply test_manager/reports/summary/summary.json:1]: Let address this after this PR. Diff is not acting as intended even its in gitignore list

**Review by billpittman — APPROVED** [review]

**Pull request merged**